src/wazuh_modules/wm_sca.c

```diff
@@ -625,6 +629,27 @@ static int wm_sca_check_policy(cJSON *policy, cJSON *profiles) {
             read_id = (int *) realloc(read_id, sizeof(int) * (i + 2));
             read_id[i] = check_id->valueint;
             read_id[i + 1] = 0;
+
+            rules_id = cJSON_GetObjectItem(check, "rules");
+
+            cJSON_ArrayForEach(rule, rules_id){
+                if (rules_id == NULL) {
+                    merror("Rules not found.");
```

Comment:
Here we know the check ID and it is valid, we can improve the log message: `merror("Invalid check %d: no rules found.", check_id->valueint);`